### PR TITLE
some fixes for complex function

### PR DESCRIPTION
**Pull request by vruge**
e.g. (a,-b) = f(.) -> (a,c) = f(.) with c = -b
       (a,b) = f(a) fixed iterration var

### CODE REVIEW

**Comment by lochel** [comment]
Please squash your commits before you merge this pull request.

**Comment by OpenModelica-Hudson** [comment]
The tests run correctly according to [OpenModelica_TEST_PULL_REQUEST 2015-07-21_12-38-29](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/407/console).

**Comment by OpenModelica-Hudson** [comment]
The tests run correctly according to [OpenModelica_TEST_PULL_REQUEST 2015-07-21_13-07-52](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/408/console).

File: Compiler/Util/Flags.mo [hunk 1]
```diff
@@ -761,6 +764,7 @@ constant ConfigFlag POST_OPT_MODULES = CONFIG_FLAG(16, "postOptModules",
     "solveLinearSystem",
     "addScaledVars",
     "removeSimpleEquations",
+    "simplifyComplexFunction",
```
Comment (lochel) [inline Compiler/Util/Flags.mo]: A (brief) description of the newly introduced optimization module "simplifyComplexFunction" is missing.

**Comment by lochel** [comment]
**This is a blocker for the new "early-initialization".**
It is critical to introduce variables during post-optimization phase since those variables will not be synchronized between different tasks (e.g. initialization and simulation) and are not generated for other systems than for simulation. It would be good to fix this more properly without introducing variables during post-optimization.

**Pull request merged**